File: test/end-to-end/Conversions/Ruby/converters.rb
#
def try_as_double(doubleLike)
    asDouble = Float(doubleLike) rescue nil

    if (asDouble != nil)
        puts "It's a double: %d" % [asDouble.floor]
    end
end

def try_as_int(firstIntLike, secondIntLike)
    firstInt = Float(firstIntLike) rescue nil
    secondInt = Float(secondIntLike) rescue nil

    if (firstInt != nil && secondInt != nil)
        puts "Both are ints: %d %d" % [firstInt, secondInt]
    end
end
#

File: test/end-to-end/Conversions/PHP/converters.php
<?php
function tryAsDouble($doubleLike) {
    if (is_numeric($doubleLike)) {$asDouble = floatval($doubleLike);

        echo 'It is a double: ' . floor($asDouble) . '' . "\n";
    }
}

function tryAsInt($firstIntLike, $secondIntLike) {
    if (ctype_digit($firstIntLike) && ctype_digit($secondIntLike)) {$firstInt = intval($firstIntLike);
    $secondInt = intval($secondIntLike);

        echo 'Both are ints: ' . $firstInt . ' ' . $secondInt . '' . "\n";
    }
}
